paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java

```diff
@@ -41,6 +43,9 @@
  */
 public abstract class FileFormat {
 
+    private static final ObjectCacheManager<String, FileFormatFactory> formatFactoryCache =
```

Comment:
Can you just make `FileFormatFactory` extends `Factory` and use `FactoryUtil.discoverFactory`?

Reply:
@JingsongLi Let me test the performance first

Reply:
@JingsongLi 
English:I have tested that using FactoryUtil.discoverFactory is still very slow, and the performance is similar to using ServiceLoader.load directly, although using FactoryUtil.discover is more unified and optimized. However, the performance gap between them and the direct use of object cache is at least 30 times. Due to the excessive number of FileFormat object creations in a single query, actual testing shows that without cache, the planning phase time will increase to about 20ms, and it will soar to 20ms under high QPS. More than 100ms, and now the average plan time of the optimization result table name is less than 3ms even when the qps is greater than 1000.
For details, see org.apache.paimon.format.FormatPerformanceTest

Chinese: 我已经测试了，使用FactoryUtil.discoverFactory依然很慢，和直接使用ServiceLoader.load的性能差不多，虽然使用FactoryUtil.discover更加统一和优化。但是他们和直接使用object cache的性能差距至少30倍，由于在单次查询中FileFormat对象创建次数过多，实际测试，不用cache的会导致plan阶段耗时增加到20ms左右，高qps下会飙升到100ms以上，而现在优化结果表名即使在qps大于1000的情况下，plan平均耗时也在3ms以下。
具体可以看org.apache.paimon.format.FormatPerformanceTest

```
0.0024763299, 0.0863487761 0.0816213514
```

Reply:
![image](https://github.com/user-attachments/assets/f6a75ef4-6503-408f-b19a-9c4a30ed6391)
Github tests is slow more than 10 times . :(

Reply:
Through your test, I obtained the completely opposite result

Reply:
In my ubuntu22.04 show as here, I do not know why , delete this future for while , put this in another pr.

```
0.0024763299, 0.0863487761 0.0816213514
```